src/routes/RequestsRoute.js

```diff
@@ -365,6 +368,7 @@ class RequestsRoute extends React.Component {
     const { mutator } = this.props;
     const { id } = this.getCurrentServicePointInfo();
     mutator.currentServicePoint.update({ id });
+    this.buildRecordsForHoldsShelfReport();
```

Comment:
I think generate report each time when component re-renders not a good idea. If it possible to find a condition when it shout be done?